manager/controlapi/node.go

```diff
@@ -136,6 +136,10 @@ func (s *Server) ListNodes(ctx context.Context, request *api.ListNodesRequest) (
 				if len(request.Filters.Labels) == 0 {
 					return true
 				}
+				if filterMatchLabels(e.Spec.Annotations.Labels, request.Filters.Labels) == true {
```

Comment:
I think that for the SwarmKit side, this should be a separate filter, or at least a way to distinguish between "node" labels and "engine" labels.

We can still decide to keep the "docker" UX the same (filtering by labels using both "engine" and "node" labels), but at least it gives control over what is filtered on at the API level.

@aluzzardi @allencloud wdyt?